### PR TITLE
Reform log

### PR DESCRIPTION
**Pull request by kubazak**
I have modified some logs to meet conventions. In class perun.core.impl.AttributesManagerImpl.
I created new info logs in class perun.core.AttributesManagerBlImpl.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 1]
```diff
@@ -3689,6 +3689,7 @@ public boolean removeAttributeWithoutCheck(PerunSession sess, String key, Attrib
 			} catch (WrongAttributeValueException | WrongReferenceAttributeValueException ex) {
 				throw new InternalErrorException(ex);
 			}
+			log.info("{} removed attribute: {} by key = {}",sess.getPerunClient().getId(), attribute.getName(), key);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: ``PerunClient`` is valid only for OAUTH2 clients. Otherwise its half empty object, especially client ID is not present. I would prefer logging whole ``PerunSession``, but I understand, that this message should be rather short. But neither ``PerunSession`` or ``PerunPrincipal`` within it have a short ``toString()``. Please add method in ``PerunPrincipal`` like: ``getLogId()`` which will return value of ``actor`` property, so we will see user identity. If it won't be sufficient in a future, we can safely change such method to return also extSourceName or other properties. Thank you.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 2]
```diff
@@ -4131,6 +4141,7 @@ public boolean removeAttributeWithoutCheck(PerunSession sess, Resource resource,
 			throw new InternalErrorException(ex);
 		}
 		if (changed)
+			log.info("{} removed attribute({}) from esource. Id resource = {}",sess.getPerunClient().getId(), attribute.getName(), resource.getId());
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: ``esource`` -> ``resource``
Also you probably don't want () around {} which is replaced for attr name, since in other messages you don't do that.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 3]
```diff
@@ -4335,15 +4335,15 @@ public void convertAttributeValuesToUnique(PerunSession session, AttributeDefini
 								break;
 						}
 						int c = counter.addAndGet(1);
-						if(c%1000==0) log.info(""+c+" values of "+attrDef.getName()+" were converted");
+						if(c%1000==0) log.debug(""+c+" values of "+attrDef.getName()+" were converted");
```
Comment (martin-kuba) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: Do not use string concatenation for constructing log messages, use message formatting with {} as placeholders and variable arguments for values instead. Message formatting is more efficient if the active loglevel is less detailed than the level of this statement, because no new objects need to be created in that case.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 4]
```diff
@@ -3451,7 +3451,7 @@ public boolean removeVirtualAttribute(PerunSession sess, Resource resource, Grou
 	public boolean removeAttribute(PerunSession sess, User user, AttributeDefinition attribute) throws InternalErrorException {
 		try {
 			if (0 < jdbc.update("DELETE FROM user_attr_values WHERE attr_id=? AND user_id=?", attribute.getId(), user.getId())) {
-				log.info("Attribute (its value) was removed from user. Attribute={}, user={}", attribute, user);
+				log.debug("Attribute (its value) was removed from user. Attribute={}, user={}", attribute, user);
```
Comment (martin-kuba) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: "Attribute value was removed" is more concise than "Attribute (its value) was removed"

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 5]
```diff
@@ -4516,6 +4534,7 @@ public void removeAllUserFacilityAttributes(PerunSession sess, User user) throws
 
 		//remove all non-virtual attributes
 		getAttributesManagerImpl().removeAllUserFacilityAttributes(sess, user);
+		log.info("{} removed all attributes from user on all facilities. User id = ", user.getId());
```
Comment (martin-kuba) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Should be 
```
log.info("{} removed all attributes from user {} on all facilities", sess.getLogId(), user.getId());
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 6]
```diff
@@ -4293,6 +4306,7 @@ public boolean removeAttributeWithoutCheck(PerunSession sess, Member member, Gro
 			} catch (WrongAttributeValueException | WrongReferenceAttributeValueException ex) {
 				throw new InternalErrorException(ex);
 			}
+			log.info("{} removed this attribute({}) from member in group. Id member = {}, id group = {}",sess.getPerunClient().getId(), attribute.getName(), member.getId(), group.getId());
```
Comment (martin-kuba) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: I would prefer the format to use id next to the entity type, like
```
log.info("{} removed  attribute {} from member {} in group {}",...
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 7]
```diff
@@ -3929,6 +3933,7 @@ public boolean removeAttributeWithoutCheck(PerunSession sess, Host host, Attribu
 			} catch (WrongAttributeValueException | WrongReferenceAttributeValueException ex) {
 				throw new InternalErrorException(ex);
 			}
+			log.info("{} removed this attribute {} from  host {}", sess.getLogId(), attribute.getName(), host.getId());
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: I would remove unnecessary "this" from the message (and other similar messages). Also there is double space before "host" word.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 8]
```diff
@@ -2110,7 +2110,7 @@ public void deleteAttribute(PerunSession sess, AttributeDefinition attribute) th
 
 			jdbc.update("DELETE FROM "+ attributeToTablePrefix(attribute)+"_attr_values WHERE attr_id=?", attribute.getId());
 			jdbc.update("DELETE FROM attr_names WHERE id=?", attribute.getId());
-			log.info("Attribute deleted [{}]", attribute);
+			log.debug("Attribute deleted [{}]", attribute);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: Attribute object is unncessary enclosed in ``[ ]``, opposite to createAttribute() method/message. I suggest ``Attribute created: {}`` and ``Attribute deleted: {}``.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 9]
```diff
@@ -2731,7 +2731,7 @@ public Attribute fillAttribute(PerunSession sess, Resource resource, Attribute a
 		//Use attributes module
 		ResourceAttributesModuleImplApi attributeModule = getResourceAttributeModule(sess, attribute);
 		if (attributeModule == null) {
-			log.debug("fillAttribute - There's no attribute module for this attribute. Attribute wasn't filled. Attribute={}", attribute);
+			log.debug("fillAttribute - There's no attribute module for this attribute. Attribute {} wasn't filled.", attribute);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: Original message is not ok, but having placeholder for attribute object at the end has meaning. Since ``toString()`` on Attribute if pretty long, you won't see "wasn't filled." on first sight. Maybe rephrase whole message, like: ``Attribute wasn't filled. There is no module for: {}`` and place attribute object at the end. Also I think that attribute name is sufficient in this message, so you don't have to pass the whole object.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 10]
```diff
@@ -3317,7 +3317,7 @@ public void removeAllAttributes(PerunSession sess, Resource resource) throws Int
 	public boolean removeAttribute(PerunSession sess, Resource resource, Member member, AttributeDefinition attribute) throws InternalErrorException {
 		try {
 			if (0 < jdbc.update("DELETE FROM member_resource_attr_values WHERE attr_id=? AND member_id=? AND resource_id=?", attribute.getId(), member.getId(), resource.getId())) {
-				log.info("Attribute (its value) was removed from member on resource. Attribute={}, member={}, resource=" + resource, attribute, member);
+				log.debug("Attribute value {} was removed from member {} on resource {}" + resource, attribute, member);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: Technically its usefull to have full objects in the message, but also in this case we can't relay on previous (correct) value to be present in passed ``AttributeDefinition`` (which, in a fact, lacks value). So maybe rephrasing all message to something like this would help: ``Attribute value for {} was removed from member {} on resource {}.`` where you again pass attribute name and then member and resource objects.
Also there is an inconsistency with ending dot between this kind of messages.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 11]
```diff
@@ -1381,7 +1381,7 @@ public AttributeDefinition getAttributeDefinition(PerunSession sess, String attr
 			else if (entity instanceof UserExtSource) ues = (UserExtSource) entity;
 				//Else skip not identified entity (log it)
 			else
-				log.debug("In method GetAttributesDefinitionWithRights there are entity which is not identified correctly and will be skipped: " + entity);
+				log.warn("In method GetAttributesDefinitionWithRights there are entity which is not identified correctly and will be skipped: {}", entity);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: there are entity -> there is entity

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 12]
```diff
@@ -3882,6 +3885,7 @@ public void removeAllAttributes(PerunSession sess, Facility facility, boolean re
 		if (removeAlsoUserFacilityAttributes) {
 			List<Attribute> userFacilityAttributes = getUserFacilityAttributesForAnyUser(sess, facility);
 			getAttributesManagerImpl().removeAllUserFacilityAttributesForAnyUser(sess, facility);
+			log.info("{} removed all attributes from any user facility {}.",sess.getLogId(), facility.getId());
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: from any user facility {} -> from any user on facility {}

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 13]
```diff
@@ -6934,7 +6959,7 @@ private void updateInverseDependenciesForAttribute(Map<AttributeDefinition, Set<
 					AttributeDefinition attrDef = getAttributeDefinition(sess, s);
 					strongDepSet.add(attrDef);
 				} catch (AttributeNotExistsException ex) {
-					log.error("For attribute dependency name " + s + "can't be found attributeDefinition.");
+					log.error("For attribute dependency name {} can't be found attributeDefinition.", s);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: I would prefer something like: ``AttributeDefinition can't be found for dependency {}.`` Same for the else branch.

File: perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java [hunk 14]
```diff
@@ -7081,7 +7106,7 @@ public void convertAttributeToUnique(PerunSession session, int attrId) throws In
 		long startTime = System.currentTimeMillis();
 		attributesManagerImpl.convertAttributeValuesToUnique(session, attrDef);
 		long endTime = System.currentTimeMillis();
-		log.info("attribute {} was converted to unique in {} ms",attrDef.getName(),(endTime-startTime));
+		log.debug("attribute {} was converted to unique in {} ms",attrDef.getName(),(endTime-startTime));
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java]: Start with big A, ends with dot.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 15]
```diff
@@ -2089,7 +2089,7 @@ public void deleteAttribute(PerunSession sess, AttributeDefinition attribute) th
 	public void deleteAllAttributeAuthz(PerunSession sess, AttributeDefinition attribute) throws InternalErrorException {
 		try {
 			if (0 < jdbc.update("DELETE FROM attributes_authz WHERE attr_id=?", attribute.getId())) {
-				log.info("All attribute_authz were deleted for Attribute={}.", attribute);
+				log.debug("All attribute_authz were deleted for Attribute={}.", attribute);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: Since you log the whole object you can change ending to ``deleted for {}.``, since attributes ``toString()`` is like ``Attribute:[content]``.

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java [hunk 16]
```diff
@@ -3394,7 +3394,7 @@ public void removeAllUserFacilityAttributesForAnyUser(PerunSession sess, Facilit
 	public void removeAllUserFacilityAttributes(PerunSession sess, User user) throws InternalErrorException {
 		try {
 			if (0 < jdbc.update("DELETE FROM user_facility_attr_values WHERE user_id=?", user.getId())) {
-				log.info("All attributes (theirs values) were removed from user on  all facilities. User={}", user);
+				log.debug("All attributes s values were removed from user {} on  all facilities.", user);
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/AttributesManagerImpl.java]: All attributes s values were -> All attributes values were
